Applications/SpaceAnalyzer/main.cpp

```diff
@@ -0,0 +1,308 @@
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "TreeMapWidget.h"
+#include <AK/Queue.h>
+#include <AK/QuickSort.h>
+#include <AK/RefCounted.h>
+#include <Applications/SpaceAnalyzer/SpaceAnalyzerGML.h>
+#include <LibCore/DirIterator.h>
+#include <LibCore/File.h>
+#include <LibGUI/AboutDialog.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/BreadcrumbBar.h>
+#include <LibGUI/Icon.h>
+#include <LibGUI/Menu.h>
+#include <LibGUI/MenuBar.h>
+#include <LibGUI/StatusBar.h>
+#include <sys/stat.h>
+
+static const char* APP_NAME = "SpaceAnalyzer";
+
+struct TreeNode : public TreeMap::ITreeMapNode {
+    TreeNode(String name)
+        : m_name(name)
+        , m_area(0)
+        , m_children(nullptr) {};
+    String m_name;
+    int64_t m_area;
+    OwnPtr<Vector<TreeNode>> m_children;
+
+    virtual String name() { return m_name; }
+    virtual int64_t area() { return m_area; }
+    virtual size_t num_children()
+    {
+        if (m_children)
+            return m_children->size();
+        return 0;
+    }
+    virtual TreeNode& child_at(size_t i) { return m_children->at(i); }
+    virtual void sort_children_by_area()
+    {
+        if (m_children != nullptr) {
+            quick_sort(*m_children, [](auto& a, auto& b) { return b.m_area < a.m_area; });
+        }
+    }
+};
+
+struct Tree : public TreeMap::ITreeMap {
+    Tree(String root_name)
+        : m_root(root_name) {};
+    virtual ~Tree() {};
+    TreeNode m_root;
+    virtual TreeMap::ITreeMapNode* get_root() override
+    {
+        return &m_root;
+    };
+};
+
+struct MountInfo {
+    String mount_point;
+    String source;
+};
+
+static void fill_mounts(Vector<MountInfo>& output)
+{
+    // Output info about currently mounted filesystems.
+    auto df = Core::File::construct("/proc/df");
+    if (!df->open(Core::IODevice::ReadOnly)) {
+        fprintf(stderr, "Failed to open /proc/df: %s\n", df->error_string());
+        return;
+    }
+
+    auto content = df->read_all();
+    auto json = JsonValue::from_string(content);
+    ASSERT(json.has_value());
+
+    json.value().as_array().for_each([&output](auto& value) {
+        auto fs_object = value.as_object();
+        MountInfo mi;
```

Comment:
Please use full words instead of abbreviations in variable names. (See [coding style](https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#names))